src/sidebar/util/root-thread.js

```diff
@@ -0,0 +1,75 @@
+import buildThread from './build-thread';
+import memoize from './memoize';
+import * as metadata from './annotation-metadata';
+import { generateFacetedFilter } from './search-filter';
+import filterAnnotations from './view-filter';
+import { shouldShowInTab } from './tabs';
+
+/** @typedef {import('./build-thread').Thread} Thread */
+/** @typedef {import('./build-thread').Options} BuildThreadOptions */
+/** @typedef {import('../store/modules/selection').ThreadState} ThreadState */
+
+// Sort functions keyed on sort option
+const sortFns = {
+  Newest: function (a, b) {
+    return a.updated > b.updated;
+  },
+  Oldest: function (a, b) {
+    return a.updated < b.updated;
+  },
+  Location: function (a, b) {
+    return metadata.location(a) < metadata.location(b);
+  },
+};
+
+/**
+ * Cobble together the right set of options and filters based on current
+ * `threadState` to build the root thread.
+ *
+ * @param {ThreadState} threadState
+ * @return {Thread}
+ */
+function buildRootThread(threadState) {
+  const selection = threadState.selection;
+
+  /** @type {BuildThreadOptions} */
+  const options = {
+    expanded: selection.expanded,
+    forcedVisible: selection.forcedVisible,
+    selected: selection.selected,
+    sortCompareFn: sortFns[selection.sortKey],
+  };
+
+  // Is there a filter query present, or an applied user (focus) filter?
+  // If so, we'll need to filter the annotations
+  const annotationsFiltered =
+    !!selection.filterQuery || Object.keys(selection.filters).length > 0;
+
+  if (annotationsFiltered) {
+    const filters = generateFacetedFilter(
+      selection.filterQuery,
+      selection.filters
+    );
+    options.filterFn = ann => filterAnnotations([ann], filters).length > 0;
+  }
+
+  // If annotations aren't filtered, should we filter out tab-irrelevant
+  // annotations (e.g. we should only show notes in the `Notes` tab)
+  // in the sidebar?
+  const threadFiltered =
+    !annotationsFiltered && threadState.route === 'sidebar';
+
+  if (threadFiltered) {
+    options.threadFilterFn = thread => {
+      if (!thread.annotation) {
+        return false;
+      }
+      return shouldShowInTab(thread.annotation, selection.selectedTab);
+    };
+  }
+  return buildThread(threadState.annotations, options);
+}
+
+const thread = memoize(buildRootThread);
+
+export default thread;
```

Comment:
Even though `thread` is a valid verb, I think it would be clearer to use a more verbose name when seeing this function in other files - eg. `buildRootThread(store.threadState())` or `createRootThread(store.threadState())`.

Also, to match other modules if a module default-exports a function then I suggest the module name should be the kebab-cased version of the function name (eg. similar to how `build-thread.js` exports `buildThread` and `use-store.js` exports `useStore`).

Reply:
I agree that this name isn't descriptive enough. Will change.